aesd-char-driver/main.c

```diff
@@ -18,58 +18,202 @@
 #include <linux/cdev.h>
 #include <linux/fs.h> // file_operations
 #include "aesdchar.h"
+#include "aesd-circular-buffer.h"
+#include <linux/slab.h>
+
 int aesd_major =   0; // use dynamic major
 int aesd_minor =   0;
 
-MODULE_AUTHOR("Your Name Here"); /** TODO: fill in your name **/
+MODULE_AUTHOR("Deekshith Reddy Patil"); /** TODO: fill in your name **/
 MODULE_LICENSE("Dual BSD/GPL");
 
 struct aesd_dev aesd_device;
 
 int aesd_open(struct inode *inode, struct file *filp)
 {
 	PDEBUG("open");
+
 	/**
 	 * TODO: handle open
 	 */
+
```

Comment:
@DeekshithPatil 	You are missing a task here
         //fetching pointer to the containing structure
	dev = container_of(inode->i_cdev, struct aesd_dev, cdev);
	//store the returned pointer to private data
	filp->private_data = dev;